jgiven-html5-report/src/main/java/com/tngtech/jgiven/report/html5/Html5AttachmentGenerator.java

```diff
@@ -195,4 +201,78 @@ static BufferedImage scaleDown( BufferedImage before ) {
         }
         return imageArray;
     }
+
+    private void writeThumbnailForSVG(File initialSVG, File thumbnailSVG) {
+        String base64PNGImage = getPNGFromSVG(initialSVG);
+        byte[] scaledDownInBytes = compressToThumbnail(base64PNGImage, "png");
+        Dimension imageDimension = getImageDimension(scaledDownInBytes);
+        String base64ScaledDownContent = BaseEncoding.base64().encode(scaledDownInBytes);
+        createSVGThumbFile(thumbnailSVG, base64ScaledDownContent, imageDimension);
+    }
+
+    Dimension getImageDimension(byte[] givenImage) {
+        ByteArrayInputStream bais = new ByteArrayInputStream(givenImage);
+        Dimension dimension = new Dimension();
+        try {
+            BufferedImage image = ImageIO.read(bais);
+            dimension.height = image.getHeight();
+            dimension.width = image.getWidth();
+        } catch (IOException e) {
+            log.error("The converted png image cannot be read.");
+        }
+
+        return dimension;
+    }
+
+    void createSVGThumbFile(File targetFile, String base64Image, Dimension dimension) {
+        String xmlFormat = getXMLFormat(base64Image, dimension);
+        try {
+            FileWriter fileWriter = new FileWriter(targetFile);
+            fileWriter.write(xmlFormat);
+            fileWriter.close();
```

Comment:
This will hold the file open if an exception is thrown.

Try-with-resources might be better, or failing this try-catch-finally will do